src/benchq/compilation/rbs_hyperparam_tuning.py

```diff
@@ -0,0 +1,451 @@
+        circuit (Circuit): the circuit to run RBS on
+        circuit_prop_estimate (float): copying a large circuit to julia can take
+            a long time due to turning it into a string and back, so estimate how
+            much of the circuit RBS will get through before rbs_iteration_time
+            runs out, and supply that here to speed up optimization
+
+    Returns:
+        callable function which accepts an optuna "trial" and returns the cost
+    """
+
+    def objective(trial):
+        teleportation_threshold = trial.suggest_int("teleportation_threshold", 10, 70)
+        teleportation_distance = trial.suggest_int("teleportation_distance", 1, 7)
+        min_neighbors = trial.suggest_int("min_neighbors", 1, 11)
+        max_num_neighbors_to_search = trial.suggest_int(
+            "max_num_neighbors_to_search", 10000, 100000
+        )
+        decomposition_strategy = trial.suggest_categorical(
+            "decomposition_strategy", [0, 1]
+        )
+
+        return estimated_time_cost_from_rbs(
+            rbs_iteration_time=rbs_iteration_time,
+            circuit=circuit,
+            verbose=False,
+            max_graph_size=None,
+            teleportation_threshold=teleportation_threshold,
+            teleportation_distance=teleportation_distance,
+            min_neighbors=min_neighbors,
+            max_num_neighbors_to_search=max_num_neighbors_to_search,
+            decomposition_strategy=decomposition_strategy,
+            circuit_prop_estimate=circuit_prop_estimate,
+        )
+
+    return objective
+
+
+def get_optimal_hyperparams_for_space(
+    rbs_iteration_time: float,
+    max_allowed_time: float,
+    circuit: Circuit,
+    n_trials: int,
+    circuit_prop_estimate: float = 1.0,
+):
+    """
+    Function to get optimal hyperparameters for space complexity after running
+    RBS.
+
+    Args:
+        rbs_iteration_time (float): amount of time to let RBS run for in each trial
+        max_allowed_time (float): maximum time you want RBS to run when you
+            eventually find the whole graph
+        circuit (Circuit): the circuit to run RBS on
+        n_trials (int): total number of trials to run when optimizing
+        circuit_prop_estimate (float): copying a large circuit to julia can take
+            a long time due to turning it into a string and back, so estimate how
+            much of the circuit RBS will get through before rbs_iteration_time
+            runs out, and supply that here to speed up optimization
+
+    Returns:
+        the best hyperparameters found during optimization
+    """
+    objective = create_space_time_objective_fn(
+        rbs_iteration_time,
+        max_allowed_time,
+        "space",
+        circuit,
+        circuit_prop_estimate,
+    )
+    study = optuna.create_study()
+    study.optimize(objective, n_trials=n_trials, catch=(ValueError,))
+    # study.optimize(objective, n_trials=n_trials)
```

Comment:
Why's this code commented out? Can it be uncommented to change behavior?

Reply:
Ohhhhh that's my bad, relic from testing which I missed. Will fix!

Reply:
fixed in 377d207a5a41b899f190ed985b8db0531c054bb4